src/Microsoft.AspNet.Razor/Parser/HtmlMarkupParser.Block.cs

```diff
@@ -1043,6 +1043,12 @@ private void EndTagBlock(Stack<Tuple<HtmlSymbol, SourceLocation>> tags, bool com
             if (!Context.DesignTimeMode)
             {
                 AcceptWhile(HtmlSymbolType.WhiteSpace);
+                if (Context.LastSpan.Kind == SpanKind.Transition)
+                {
+                    // Mark the whitespace at the end of a <text> tag as code.
+                    Span.CodeGenerator = new StatementCodeGenerator();
+                    Output(SpanKind.Code);
```

Comment:
Hmm, actually in order to make this truly only affect after transition bits it'd probably be best to only `Output(SpanKind.Markup)` prior to the whitespace IF the last span happens to be a transition. This way the change doesn't affect non after transition based scenarios.